### PR TITLE
Aarch64 - loadb/movxbl  -> loadzbl peephole

### PR DESCRIPTION
**Pull request by swalk-cavium**
On Aarch64 the loadb/movxbl sequence would generate ldrsb/uxtb.
This occurred around the  Vinstr cmpbm instructions.  It is equivalent to just use 
the ldrb instruction and save one instruction per instance.

Before
====
    14854    0: FPushFuncD 0 "ut_main"
    14855     (03) EndGuards
    14856     (07) t3:Func = LdFuncCached<ut_main> -> B2<Catch>
    14857         Main:   
    14858               0x296008b0  d28217e0              movz x0, #0x10bf 
    14859               0x296008b4  38e06b60              ldrsb w0, [x27, x0]     // <<--+
    14860               0x296008b8  53001c00              uxtb w0, w0             // <<--+
    14861               0x296008bc  d29fffe1              movz x1, #0xffff 
    14862               0x296008c0  f2a05fe1              movk x1, #0x2ff, lsl #16 
    14863               0x296008c4  38616b61              ldrb w1, [x27, x1] 
    14864               0x296008c8  6b00003f              cmp w1, w0

After
=====
     320    0: FPushFuncD 0 "ut_main"
     321     (03) EndGuards
     322     (07) t3:Func = LdFuncCached<ut_main> -> B2<Catch>
     323         Main:
     324               0x1d40082c  d28217e0              movz x0, #0x10bf
     325               0x1d400830  38606b60              ldrb w0, [x27, x0]  //<<---
     326               0x1d400834  d29fffe1              movz x1, #0xffff
     327               0x1d400838  f2a05fe1              movk x1, #0x2ff, lsl #16
     328               0x1d40083c  38616b61              ldrb w1, [x27, x1]
     329               0x1d400840  6b00003f              cmp w1, w0

This was tested by running the regression tests with six different option sets.
No new issues were found.

### CODE REVIEW

File: hphp/runtime/vm/jit/vasm-simplify.cpp [hunk 1]
```diff
@@ -431,6 +431,21 @@ bool simplify(Env& env, const copyargs& inst, Vlabel b, size_t i) {
   });
 }
 
+bool simplify(Env& env, const loadb& inst, Vlabel b, size_t i) {
+  return if_inst<Vinstr::movzbl> (env, b, i+1, [&] (const movzbl& cm) {
+    // loadb; movzbl; -> loadzbl;
+    if (!(arch() == Arch::ARM &&
+          env.use_counts[inst.d] == 1 &&
+          env.use_counts[cm.s] == 1 &&
```
Comment (mxw) [inline hphp/runtime/vm/jit/vasm-simplify.cpp]: This condition is redundant; it's implied by the other two.

File: hphp/runtime/vm/jit/vasm-simplify.cpp [hunk 2]
```diff
@@ -431,6 +431,21 @@ bool simplify(Env& env, const copyargs& inst, Vlabel b, size_t i) {
   });
 }
 
+bool simplify(Env& env, const loadb& inst, Vlabel b, size_t i) {
+  return if_inst<Vinstr::movzbl> (env, b, i+1, [&] (const movzbl& cm) {
```
Comment (mxw) [inline hphp/runtime/vm/jit/vasm-simplify.cpp]: Two nits:
1. No space after function names before the paren.
2. Space between operators (`i + 1`).

File: hphp/runtime/vm/jit/vasm-simplify.cpp (hunk 1 above)
Comment (swalk-cavium) [inline hphp/runtime/vm/jit/vasm-simplify.cpp]: Hi @mxw, I'm not sure which condition you are referencing.  The arch() line?  The whole 'if'?

Comment (mxw) [inline hphp/runtime/vm/jit/vasm-simplify.cpp]: I'm referring to the specific line I've attached this comment to.

If `inst.d == cm.s`, then obviously `env.use_counts[inst.d] == 1` will imply that `env.use_counts[cm.s] == 1`.

**Pull request closed without merging**